src/socket.cr

```diff
@@ -250,7 +250,7 @@ class Socket < IO
         if closed?
           return
         elsif Errno.value == Errno::EAGAIN
-          wait_readable
+          wait_readable rescue nil
```

Comment:
@bcardiff shouldn't be this `rescue return nil` ? The loop doesn't break otherwise. We're seeing hanging on closing a socket.